### PR TITLE
Color picker

### PR DESCRIPTION
**Pull request by LuigiCerone**
# Description

Hi,

thanks again for all the useful information you provided us in issue #1622. 

@mfumanelli and I have completed a first version of the code.

To summarize the activities:
* We defined the [ColorPicker](https://github.com/LuigiCerone/gradio/blob/color-picker/gradio/components.py#L2800) component in the components.py file;
* We defined the [ColorPicker.svelte](https://github.com/LuigiCerone/gradio/tree/color-picker/ui/packages/app/src/components/ColorPicker) component on the UI side by exploiting the [type color input of HTML](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color);
* We created a demo in [demo/color_picker](https://github.com/LuigiCerone/gradio/tree/color-picker/demo/color_picker);
* We added both [Python](https://github.com/LuigiCerone/gradio/blob/color-picker/test/test_components.py#L1670) and [JS](https://github.com/LuigiCerone/gradio/blob/color-picker/ui/packages/app/src/components/ColorPicker/ColorPicker.test.ts) tests for the new components.

For now, our concerns are:
* In the [IOComponent](https://github.com/LuigiCerone/gradio/blob/color-picker/gradio/components.py#L197) we found methods about "interpretation", we don't really understand this aspect and we don't know if we need to handle it with respect to the component we implemented (and its related tests);
* For documentation we didn't understand if this is automatically generated from the docstring of the methods or if it has to be written by hand.

Thanks again for your help!

Closes: #1622

# Checklist:

- [X] I have performed a self-review of my own code
- [X] My code follows the style guidelines of this project
- [X] I have commented my code in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [x] Check if `interpretation`'s stuff is required with new component
- [x] Check if documentation generation is automatic
- [x] Run style scripts

@pngwn @FarukOzderim @abidlabs

### CODE REVIEW

**Comment by FarukOzderim** [comment]
Thx for the contribution!

> For documentation we didn't understand if this is automatically generated from the docstring of the methods or if it has to be written by hand.
It is automatically generated as far as I know.

File: test/test_components.py [hunk 1]
```diff
@@ -1667,5 +1667,105 @@ def test_in_interface(self):
...
+        """
+        color_picker_input = gr.ColorPicker()
+        self.assertEqual(color_picker_input.preprocess("#000000"), "#000000")
+        self.assertEqual(color_picker_input.preprocess_example("#000000"), "#000000")
+        self.assertEqual(color_picker_input.postprocess(None), None)
+        self.assertEqual(color_picker_input.postprocess("#FFFFFF"), "#FFFFFF")
+        self.assertEqual(color_picker_input.serialize("#000000", True), "#000000")
+
+        
+        # text_input.interpretation_replacement = "unknown"
```
Comment (FarukOzderim) [inline test/test_components.py:1683]: Why do we have commented out tests over here?

File: gradio/components.py [hunk 2]
```diff
@@ -2797,6 +2797,143 @@ def style(
...
+    @staticmethod
+    def update(
+        value: Optional[Any] = None,
+        label: Optional[str] = None,
+        show_label: Optional[bool] = None,
+        visible: Optional[bool] = None,
+        interactive: Optional[bool] = None,
+    ):
+        updated_config = {
+            "label": label,
```
Comment (FarukOzderim) [inline gradio/components.py:2856]: better have them in the same order as in function parameters

File: gradio/components.py [hunk 3]
```diff
@@ -2797,6 +2797,143 @@ def style(
...
+    def preprocess_example(self, x: str | None) -> Any:
+        """
+        Any preprocessing needed to be performed on an example before being passed to the main function.
+        """
+        if x is None:
+            return None
+        else:
+            return str(x)
+    
+    # def set_interpret_parameters(
```
Comment (FarukOzderim) [inline gradio/components.py:2887]: imo these two functions are not necessary, and you can `return self` like [here](https://github.com/gradio-app/gradio/blob/609de11ce8a0d69907d3aef33a517ae6775e5aca/gradio/components.py#L190)

cc: @abidlabs to be sure

**Review by FarukOzderim — APPROVED** [review]
python side LGTM, @aliabid94 might take a look as well to both answer your questions about IO and because I am not very familiar with the new IO component structure and might have missed some necessary steps.

BTW would be great if you could reduce the size of the gif and image with compression, the size piles up in the repo with time.

you could `git reset HEAD^numberofcommits --soft` and recommit afterwards

**Comment by abidlabs** [comment]
This is great @LuigiCerone and @mfumanelli! The Python side looks good to me as well. Like @FarukOzderim mentioned, you don't have to worry about the interpretation-related functions

**Comment by abidlabs** [comment]
Also tested & confirmed that it works both as an input and output component -- nice work!

**Review by pngwn — APPROVED** [review]
This is awesome, kudos for adding tests on both sides too. Awesome PR.

PS: If you have any feedback about how easy/ difficult contributing was and if there is anything you think we can do to makes things easier for future contributors then please let us know!

**Comment by pngwn** [comment]
Feel free to mark this as "Ready for review".

**Review by abidlabs — APPROVED** [review]

**Comment by abidlabs** [comment]
Thank you so much @mfumanelli. If you would be open to contributing the guide on contributing components, that would be superb! I think this is definitely a Guide that would benefit from being written by a user of Gradio.

Here is some info on contributing Guides that may be helpful! https://github.com/gradio-app/gradio/blob/main/guides/CONTRIBUTING.md

**Comment by abidlabs** [comment]
Hi @mfumanelli, just wanted to ask if you were still interested in contributing the Guide on how create new components? We've gotten interest in contributing components from a few other developers so, if you are interested and have the time to put something together by early next week, we'd be happy to review it and publish it!

**Comment by mfumanelli** [comment]
Hi @abidlabs! Yes we are working on it, I think we will do the PR tomorrow

**Comment by abidlabs** [comment]
Amazing, thank you!

**Pull request merged**